packages/steveo/src/runtime/workflow.ts

```diff
@@ -126,48 +127,78 @@ export class Workflow {
     // mark the flow completed if there are no more steps
     const next = this.getNextStep(current);
     if (!next) {
-      await context.repos.workflow.workflowCompleted(workflowId);
+      await context.repos.workflow.updateWorkflowCompleted(workflowId);
       return;
     }
 
-    await context.repos.workflow.stepPointerUpdate(workflowId, next.name);
+    this.logger.debug({
+      message: 'Publish next workflow step',
+      workflowId,
+      current,
+      next: next.name,
+    });
+
+    await context.repos.workflow.updateCurrentStep(workflowId, next.name);
 
     const message = this.formatStepMessage(next.name);
 
-    await this.publishInternal(message, payload);
+    await this.publishInternal(message, { workflowId, ...payload });
   }
 
   /**
    * This handles the underlying (internal) mechanics of publishing a message.
    * It does not control flow, that is expected to be handled by the callers.
    */
-  private async publishInternal<T>(
+  private async publishInternal<T extends WorkflowPayload>(
     message: string,
     payload: T | T[],
     context?: {
       key?: string;
     }
   ) {
-    const params = Array.isArray(payload) ? payload : [payload];
+    const payloadArray = Array.isArray(payload) ? payload : [payload];
+    const logger = this.logger.child({
+      publish: message,
+      workflow: this.name,
+      workflows: Array.isArray(payload)
+        ? payload.map(p => p.workflowId)
+        : [payload.workflowId],
+    });
 
     try {
-      this.logger.info(
-        `Workflow ${this.name} publish next step in sequence ${message}`
-      );
+      logger.info({
+        message: `Workflow publish internal start`,
+      });
 
-      // sqs calls this method twice
+      // TODO: Understand and document the significance of this comment (it was copypasta'd from the original task implementation)
+      // SQS calls this method twice
       await this.producer.initialize(message);
 
       await Promise.all(
-        params.map((data: T) => {
-          this.registry.emit('workflow_send', message, data);
-          return this.producer.send(message, data, context?.key, context);
+        payloadArray.map(data => {
+          try {
+            this.registry.emit('workflow_send', message, data);
+
+            return this.producer.send(message, data, context?.key, context);
+          } catch (err) {
+            logger.info({
+              message: 'Workflow publish workflow step',
+              workflowId: data.workflowId,
+            });
+            throw err;
+          }
         })
       );
 
       this.registry.emit('workflow_success', message, payload);
+
+      logger.debug({
```

Comment:
was this meant to be a debug, we log the start as `info` above and success as `debug` here.

Reply:
wasn't my code, just following what was already there... .but i think it makes sense in order to reduce the verbosity of the info log